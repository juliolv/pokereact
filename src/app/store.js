import { configureStore } from '@reduxjs/toolkit';
import pokemonReducer from '../features/pokedex/pokemonSlice';

export default configureStore({
  reducer: {
    pokemons: pokemonReducer,
  },
});
