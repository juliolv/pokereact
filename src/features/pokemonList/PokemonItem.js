import React from 'react';
import { useDispatch } from 'react-redux';
import { release } from '../capturePokemon/pokemonSlice';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  title: {
    flexGrow: 1,
  },
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  cardMedia: {
    paddingTop: '50%',
    backgroundSize: 96,
  },
  cardContent: {
    flexGrow: 1,
    paddingTop: theme.spacing(0),
    paddingBottom: theme.spacing(1),
  },
}));

const PokemonItem = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const { id, name, img } = props.pokemon;

  const releasePokemon = (id) => {
    dispatch(release(id));
  };

  return (
    <Grid item key={id} xs={2}>
      <Card className={classes.card}>
        <CardMedia className={classes.cardMedia} image={img} title={name} />
        <CardContent className={classes.cardContent}>
          <Typography gutterBottom variant="h5" component="h2">
            {name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            Order: {id}
          </Typography>
        </CardContent>
        <CardActions>
          <Button
            size="small"
            color="primary"
            onClick={() => releasePokemon(id)}
          >
            Release
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export default PokemonItem;
