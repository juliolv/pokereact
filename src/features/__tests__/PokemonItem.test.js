import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../app/store';
import PokemonItem from '../pokedex/PokemonItem';

afterEach(() => {
  cleanup();
});

const pokemon = {
  id: 1,
  name: 'bulbasaur',
  img:
    'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
};

test('should render PokemonItem component', () => {
  render(
    <Provider store={store}>
      <PokemonItem pokemon={pokemon} />
    </Provider>
  );
  const pokemonItemElement = screen.getByTestId('pokemon-1');
  expect(pokemonItemElement).toBeInTheDocument();
  expect(pokemonItemElement).toHaveTextContent('bulbasaur');
});

test('matches snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <PokemonItem pokemon={pokemon} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
