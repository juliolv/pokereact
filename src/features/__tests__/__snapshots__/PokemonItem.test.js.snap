// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  data-testid="pokemon-1"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root makeStyles-cardMedia-4"
      style={
        Object {
          "backgroundImage": "url(\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png\\")",
        }
      }
      title="bulbasaur"
    />
    <div
      className="MuiCardContent-root makeStyles-cardContent-5"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        bulbasaur
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Order: 
        1
      </p>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Release
        </span>
      </button>
    </div>
  </div>
</div>
`;
